cly/console.py

```diff
@@ -220,7 +220,9 @@ def __init__(self, *args, **kwargs):
     }
 
     def decode(self, input, errors='strict'):
-        return _decode_re.sub(self._decode_match, str(input, encoding='ascii'))
+        if not isinstance(input, str):
```

Comment:
@guyco-redis what kind of input does it expects to get? im assuming this handling is done for unicode, right?

Reply:
it receives `bytes` which == `str` in python 2 (so no conversion is needed, and in fact, raises an error if done)
but in python 3 we need to convert it to `str` in order to perform the `sub`

Reply:
@guyco-redis I see, don't you prefer to strict it to `bytes`? this might find other issues when we try to convert a non-string to string.

Reply:
Do you mean something like this:
```
if not isinstance(input, str):
    if not isinstance(input, bytes):
        raise Exception
    input = str(input, encoding='ascii')
``` 
I'm not quite sure. On one hand this seems more safe. on the other, it means a bit more ugly, since I expect we will have lots of similar conditions (in the cnm code, where we have similar scenarios). 
@gavrie WDYT ?

Reply:
@guyco-redis maybe, since this an external library we don't have to do the same thing as cnm and we should keep that conversation somewhere else.
if we feel its safe enough we can keep it as is as long as we're sure that the outcome in py3 will be the same as py2.

Reply:
Why not simply `input.decode('ascii')`? It will work in both Python 2 (converting `bytes`/`str` to `unicode`) and Python 3 (convering `bytes` to `str`).

Reply:
I got confused with another method named `decode`. This one accepts `Anystr` which means `bytes` or `str`. so doing `input.decode('ascii')` with no condition won't work